net/NetUtil.hpp

```diff
@@ -28,6 +32,49 @@ struct sockaddr;
 namespace net
 {
 
+class Defaults
+{
+public:
+    /// WebSocketHandler ping timeout in us (2s default). Zero disables metric.
+    std::chrono::microseconds WSPingTimeout;
+    /// WebSocketHandler ping period in us (3s default), i.e. duration until next ping. Zero disables metric.
+    std::chrono::microseconds WSPingPeriod;
+    /// http::Session timeout in us (30s default). Zero disables metric.
+    std::chrono::microseconds HTTPTimeout;
+
+    /// Socket maximum connections (100000). Zero disables metric.
+    size_t MaxConnectionCount;
+    /// Socket maximum duration in seconds (12h). Zero disables metric.
+    std::chrono::seconds MaxDuration;
+    /// Socket minimum bits per seconds throughput (0). Zero disables metric.
+    double MinBytesPerSec;
+
+    /// Socket poll timeout in us (64s), useful to increase for debugging.
+    std::chrono::microseconds SocketPollTimeout;
+
+private:
+    Defaults()
+        : WSPingTimeout(std::chrono::milliseconds(2000))
+        , WSPingPeriod(std::chrono::milliseconds(3000))
+        , HTTPTimeout(std::chrono::milliseconds(30000))
+        , MaxConnectionCount(100000)
+        , MaxDuration(std::chrono::seconds(43200))
+        , MinBytesPerSec(0.0)
+        , SocketPollTimeout(std::chrono::seconds(64))
+    {
+    }
+
+public:
+    Defaults(const Defaults&) = delete;
+    Defaults(Defaults&&) = delete;
+
+    static Defaults& get()
+    {
+        static Defaults def;
```

Comment:
Default values are compile-time constants (i.e. `constexpr`). This is a pessimization and overkill for something trivial (there is a hidden lock here, btw... overkill, as I say).

And why would we want to support changing default values?! Best to keep these concepts apart: defaults are compile-time const, and runtime values are class members (variables). Unless the suggestion is to have defaults to initialize these defaults!

Please see common/Common.hpp for other default values and how we use them.

Reply:
Renamed to Config as it is a runtime configuration as required for unit testing of this feature as mentioned.